### ISSUE TITLE
Ruby/Python binding for "unround" feature

### ISSUE THREAD

**Issue by klayoutmatthias** [klayoutmatthias/klayout#2]
The "unround" feature is the basis of the capability to "edit" corner radii of polygons. The C++ function is "extract_rad" from dbPolygonTools.h. A Ruby/Python binding is asked to make this function accessible to scripts.

The proposed binding is:

    polygon = ... # some RBA::Polygon or RBA::DPolygon
    (new_polygon, outer_radius, inner_radius, n) = polygon.extract_rad

The result will be:
- new_polygon: a polygon representing the "unrounded" (corner rounding removed) version of the original polygon.
- outer_radius: the outer radius detected from the polygon
- inner_radius: the inner corner radius detected from the polygon
- n: the number of vertex points per full circle detected from the original polygon

So basically

    new_polygon.round_corners(outer_radius, inner_radius, n)

will (more or less) render the original polygon.

**Comment by klayoutmatthias** [comment]
I addition what is mentioned in the ticked, there now also are DPolygon#round_corners, DSimplePolygon#round_corners methods.

**Comment by klayoutmatthias** [comment]
This is how you can use the feature in DRC

    class DRC::DRCLayer
      def change_radius(ri, ro, n)
        res = []
        self.data.each do |p|
          x = p.extract_rad
          if x.empty?
            res << p
          else
            res << x[0].round_corners(ri / @engine.dbu, ro / @engine.dbu, n)
          end
        end
        return DRC::DRCLayer::new(@engine, RBA::Region::new(res))
      end
    end
    
    org = input(24)
    
    # Apply 1 (inner) and 2 (outer) corner rounding with 100 points/circle
    org_r1 = org.rounded_corners(1.um, 2.um, 100)
    org_r1.output(1000)
    
    # Change that to 2/3/200 
    org_r2 = org_r1.change_radius(2.um, 3.um, 200)
    org_r2.output(1001)
